### PR TITLE
fix: [CDS-82862]: Added validation for CV configs

### PR DESCRIPTION
**Pull request by ShubhenduP**
Added validation for CV configs to check if there exists a CV valid health source in all health sources

## Describe your changes
Added a validation check to verify if there exists a health source which has CV enabled in all health sources before running verify step. If it doesn't exist, skip the verify step.

## Checklist
- [ ] I've documented the changes in the PR description.
- [ ] I've tested this change either in PR or local environment.
- [ ] If hashcheck failed I followed [the checklist](https://harness.atlassian.net/wiki/spaces/DEL/pages/21016838831/PR+Codebasehash+Check+merge+checklist) and updated this PR description with my answers to make sure I'm not introducing any breaking changes.

## Comment Triggers
<details>
  <summary>Build triggers</summary>

- Feature build: `trigger feature-build`
- 
  Specific builds
  - `trigger manager`
  - `trigger dms`
  - `trigger ng_manager`
  - `trigger cvng `
  - `trigger cmdlib`
  - `trigger template_svc`
  - `trigger events_fmwrk_monitor`
  - `trigger event_server`
  - `trigger change_data_capture`
  -  Trigger multiple builds together: For eg: `trigger dms, manager`
- Immutable delegate `trigger publish-delegate`
</details>

## [Latest PR Check Triggers](https://github.com/harness/harness-core/blob/develop/.github/pull_request_template.md)

<details>
  <summary>PR Check triggers</summary>
You can run multiple PR check triggers by comma separating them in a single comment. e.g. `trigger ut0, ut1`

- Compile: `trigger compile`
- CodeformatCheckstyle: `trigger checkstylecodeformat`
  - CodeFormat: `trigger codeformat`
  - Checkstyle: `trigger checkstyle`
- MessageMetadata: `trigger messagecheck`
- File-Permission-Check: `trigger checkpermission`
- Recency: `trigger recency`
- BuildNumberMetadata: `trigger buildnum`
- Trigger CommonChecks: `trigger commonchecks`
- PMD: `trigger pmd`
- Copyright Check: `trigger copyrightcheck`
- Feature Name Check: `trigger featurenamecheck`
- UnitTests-ALL: `trigger uts`
- UnitTests-0: `trigger ut0`
- UnitTests-1: `trigger ut1`
- UnitTests-2: `trigger ut2`
- UnitTests-3: `trigger ut3`
- UnitTests-4: `trigger ut4`
- UnitTests-5: `trigger ut5`
- UnitTests-6: `trigger ut6`
- UnitTests-7: `trigger ut7`
- UnitTests-8: `trigger ut8`
- UnitTests-9: `trigger ut9`
- CodeBaseHash: `trigger codebasehash`
- CodeFormatCheckstyle: `trigger checkstylecodeformat`
- SonarScan: `trigger ss`
- GitLeaks: `trigger gitleaks`
- Trigger all Checks: `trigger smartchecks`
- Go Build: `trigger gobuild`
- Validate_Reviews: `trigger review`
- Module Dependency Check: `trigger mdc`
</details>

## PR check failures and solutions
https://harness.atlassian.net/wiki/spaces/BT/pages/21106884744/PR+Checks+-+Failures+and+Solutions

### Rebase Instructions:
To rebase your branch onto the latest changes in the target branch, simply use the following command in a comment: `/rebase`

## [Contributor license agreement](https://github.com/harness/harness-core/blob/develop/CONTRIBUTOR_LICENSE_AGREEMENT.md)

- - -
This change is [Reviewable](https://reviewable.io/reviews/harness/harness-core/55559)

### CODE REVIEW

File: srm-service/modules/cv-nextgen-service/service/src/main/java/io/harness/cvng/cdng/services/impl/CVNGStep.java [hunk 1]
```diff
@@ -400,6 +400,14 @@ private DeploymentActivity getDeploymentActivity(CVNGStepParameter stepParameter
     return deploymentActivity;
   }
 
+  private boolean hasInvalidCVConfig(List<CVConfig> cvConfigs) {
+    for (CVConfig cvConfig : cvConfigs) {
+      if (cvConfig.isDeploymentVerificationEnabled()) {
```
Comment (sribalij) [inline srm-service/modules/cv-nextgen-service/service/src/main/java/io/harness/cvng/cdng/services/impl/CVNGStep.java:405]: @ShubhenduP ,
Can you pls help me understand how the `isDeploymentVerificationEnabled` trasnslates to invalid cvConfig?

File: srm-service/modules/cv-nextgen-service/service/src/main/java/io/harness/cvng/cdng/services/impl/CVNGStep.java [hunk 2]
```diff
@@ -400,6 +400,14 @@ private DeploymentActivity getDeploymentActivity(CVNGStepParameter stepParameter
     return deploymentActivity;
   }
 
+  private boolean hasInvalidCVConfig(List<CVConfig> cvConfigs) {
```
Comment (sribalij) [inline srm-service/modules/cv-nextgen-service/service/src/main/java/io/harness/cvng/cdng/services/impl/CVNGStep.java:403]: Can you also pls add Junits

File: srm-service/modules/cv-nextgen-service/service/src/main/java/io/harness/cvng/cdng/services/impl/CVNGStep.java (hunk 1 above)
Comment (ShubhenduP) [reply srm-service/modules/cv-nextgen-service/service/src/main/java/io/harness/cvng/cdng/services/impl/CVNGStep.java:405]: This check is added for all those cases where if all the health sources are configured for either SLI or Service health. and not CV. In that case we need to skip the verify step in the pipeline, so as in the code, if any of the cvConfigs have isDeploymentVerificationEnabled(i.e. Continuous Verification) true, that means we need to run the verify step and its not an invalid config, thus returning false.

**Comment by ShubhenduP** [comment]
trigger uts

**Comment by ShubhenduP** [comment]
trigger recency

**Pull request closed without merging**